deepspeed/launcher/runner.py

```diff
@@ -390,7 +390,7 @@ def main(args=None):
     args = parse_args(args)
 
     # For when argparse interprets remaining args as a single string
-    args.user_args = shlex.split(" ".join(list(map(lambda x: x if x.startswith("-") else f"'{x}'", args.user_args))))
+    args.user_args = shlex.split(" ".join(list(map(lambda x: x if x.startswith("-") else f'"{x}"', args.user_args))))
```

Comment:
I think this would just be exchanging one problem for another. For example a `--prompt 'I am 72" tall'` would encounter the problem with these changes. I think the correct thing would be to escape any `'` characters: `{x.replace("'", "\'")}`, but I have not tested if this works.

Reply:
The outer wrapper for prompt should be double quotation (")
But I get your point, this was my first approach before finding out that inner quotations are ignored.

In other words, it ignores double quotes in the command. seems it's a known issue. so deepspeed will never see " as part of the prompt

For example: 
--prompt "He said "Hello,"" 
will be caught as: 
['--prompt', 'He said Hello,']

Reply:
I see, thank you for the explanation. In that case, I think this solution is better than what we currently have.